src/main/scala/edu/ie3/simona/service/primary/PrimaryServiceProxy.scala

```diff
@@ -412,7 +348,7 @@ case class PrimaryServiceProxy(
     * @return
     *   The [[ActorRef]] to the spun off actor
     */
-  private def classToWorkerRef[V <: Value](
+  private[primary] def classToWorkerRef[V <: Value](
```

Comment:
What is the reason for package privacy?

Reply:
Because it's now directly mocked in PrimaryServiceProxySpec:
https://github.com/ie3-institute/simona/blob/a18bac454afff36661fdd9f50b4cc95a556555e5/src/test/scala/edu/ie3/simona/service/primary/PrimaryServiceProxySpec.scala#L459-L464

Reply:
One could also make it `protected` here, I was just adapting it from the now deleted `private[primary] def columnSchemeToActor()`

Reply:
Hm... Okay... Well... Not too nice, what I introduced back then. However, fully omitting it, would require some refactoring of the primary service proxy, so I would be okay with kinda leaving it as it is, atm.

If possible, I'd go with protected here and for `initializeWorker`, as it is more restrictive. And, you might do me a favor and set `handleCoveredModel` to private.

Reply:
`classToWorkerRef` is now `protected`.

`initializeWorker` and `handleCoveredModel` cannot be made `protected` here since it is called directly in `PrimaryServiceProxySpec` (unless you'd want to introduce private method testers here).

Reply:
I'd rather go with private method testers than having the scope too loose.

Reply:
Ok, I tried both.

We cannot make `handleCoveredModel` private since `PrimaryServiceProxy` is subclassed in `PrimaryServiceProxySpec.scala:616`. Private method tester uses `getClass.getDeclaredMethods`, which means that only methods at the lowest level (here it is only `initializeWorker`) can be called. This means that we cannot call `handleCoveredModel` using private method tester.

For `initializeWorker`, there's a very similar problem to this, as we have a sublcass in `PrimaryServiceProxySpec.scala:459`.

So in short, I don't see any easy solution for making these methods private or protected.

Reply:
Hold on, found a solution just when I finished typing. Let me know what you think about it.